handlers/handlers.go

```diff
@@ -246,11 +249,85 @@ func HandleNoteApiRequest(
 
 		fmt.Fprint(responseWriter, string(notesInJson))
 
+	case http.MethodPost:
+		type NoteForm struct {
+			Content string `json:"content"`
+		}
+
+		noteForm := new(NoteForm)
+
+		if err := json.NewDecoder(request.Body).Decode(noteForm); err != nil {
+			http.Error(responseWriter, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		if len(strings.TrimSpace(noteForm.Content)) == 0 {
+			http.Error(responseWriter, "Note content cannot be empty or just whitespace", http.StatusBadRequest)
+			return
+		}
+
+		note := &models.Note{
+			AuthorId:     models.UserId(userId),
+			Content:      noteForm.Content,
+			CreationTime: time.Now().UTC(),
+		}
+
+		noteId, err := noteservice.StoreNewNote(note)
+		if err != nil {
+			http.Error(responseWriter, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		responseWriter.Write([]byte(`{NoteId: "` + strconv.FormatInt(int64(noteId), 10) + `"}`))
+		responseWriter.WriteHeader(http.StatusCreated)
+
 	default:
-		respondWithMethodNotAllowed(responseWriter, http.MethodGet)
+		respondWithMethodNotAllowed(responseWriter, http.MethodGet, http.MethodPost)
 	}
 }
 
+func HandleNoteCateogryApiRequest(
+	responseWriter http.ResponseWriter,
+	request *http.Request,
+	userId models.UserId,
+) {
+	switch request.Method {
+	case http.MethodPut:
+
+		id, err := strconv.ParseInt(request.URL.Query().Get("id"), 10, 64)
```

Comment:
We should verify that `id` is the one and only query param. And verify that it is an integer. If either of those things aren't true, we should return a 400. Right now, for some of those things, we're returning a 500

Reply:
I'll add this to our Technical backlog. Because i think eventually creating a method like respondWithMethodNotAllowed is the right call.

Reply:
https://trello.com/c/Oz9QbNVj/57-add-call-to-make-sure-there-are-not-extra-query-paramters

Reply:
I wouldn't respond with MethodNotAllowed, but other than that, I agree